manim/mobject/types/vectorized_mobject.py

```diff
@@ -1060,15 +1067,25 @@ def get_points_defining_boundary(self):
         # Probably returns all anchors, but this is weird regarding  the name of the method.
         return np.array(list(it.chain(*[sm.get_anchors() for sm in self.get_family()])))
 
-    def get_arc_length(self, n_sample_points=None):
-        if n_sample_points is None:
-            n_sample_points = 4 * self.get_num_curves() + 1
-        points = np.array(
-            [self.point_from_proportion(a) for a in np.linspace(0, 1, n_sample_points)]
+    def get_arc_length(self, sample_points_per_curve: Optional[float] = None) -> float:
+        """Return the approximated length of the whole curve.
+
+        Parameters
+        ----------
+        sample_points_per_curve
+            Number of sample points per curve used to approximate the length. More points result in a better approximation.
+
+        Returns
+        -------
+        float
+            The length of the :class:`VMobject`.
+        """
+        kwargs = {}
+        if sample_points_per_curve is not None:
+            kwargs["sample_points"] = sample_points_per_curve
+        return np.sum(
+            [length for _, length in self.get_curve_functions_with_lengths(**kwargs)]
```

Comment:
You don't need to make a new list
```suggestion
            length for _, length in self.get_curve_functions_with_lengths(**kwargs)
```